tests/kernel/threads/usage/src/task.c

```diff
@@ -0,0 +1,384 @@
+		      THREAD1_PRIO + 2, rv);
+
+	/* Raise the priority of the current thread (thread1) */
+	k_thread_priority_set(thread1_id, THREAD1_PRIO - 2);
+	rv = k_thread_priority_get(k_current_get());
+	zassert_equal(rv, THREAD1_PRIO - 2,
+		      "Expected priority to be changed to %d, not %d\n",
+		      THREAD1_PRIO - 2, rv);
+
+	/* Restore the priority of the current thread (thread1) */
+	k_thread_priority_set(thread1_id, THREAD1_PRIO);
+	rv = k_thread_priority_get(k_current_get());
+	zassert_equal(rv, THREAD1_PRIO,
+		      "Expected priority to be changed to %d, not %d\n",
+		      THREAD1_PRIO, rv);
+
+	/* Lower the priority of thread2 */
+	k_thread_priority_set(thread2_id, THREAD2_PRIO + 2);
+	k_sem_give(&sem_thread2);
+	k_sem_take(&sem_thread1, K_FOREVER);
+	zassert_equal(thread2_data, THREAD2_PRIO + 2,
+		      "Expected priority to be changed to %d, not %d\n",
+		      THREAD2_PRIO + 2, thread2_data);
+
+	/* Raise the priority of thread2 */
+	k_thread_priority_set(thread2_id, THREAD2_PRIO - 2);
+	k_sem_give(&sem_thread2);
+	k_sem_take(&sem_thread1, K_FOREVER);
+	zassert_equal(thread2_data, THREAD2_PRIO - 2,
+		      "Expected priority to be changed to %d, not %d\n",
+		      THREAD2_PRIO - 2, thread2_data);
+
+	/* Restore the priority of thread2 */
+	k_thread_priority_set(thread2_id, THREAD2_PRIO);
+	k_sem_give(&sem_thread2);
+	k_sem_take(&sem_thread1, K_FOREVER);
+	zassert_equal(thread2_data, THREAD2_PRIO,
+		      "Expected priority to be changed to %d, not %d\n",
+		      THREAD2_PRIO, thread2_data);
+}
+
+/**
+ *
+ * @brief thread2 portion to test k_sleep()
+ *
+ */
+
+void thread2_sleep_test(void)
+{
+	int  first_tick;
+
+	k_sem_take(&sem_thread2, K_FOREVER);
+
+	first_tick = _tick_get_32();
+	while (!is_main_thread_ready) {
+		/* busy work */
+	}
+	thread2_data = _tick_get_32() - first_tick;
+
+	k_sem_give(&sem_thread1);
+}
+
+/**
+ *
+ * @brief Test k_sleep()
+ *
+ */
+
+void thread_sleep_test(void)
+{
+	int  tick;
+
+	k_sem_give(&sem_thread2);
+
+	/* align on tick boundary and get current tick */
+	tick = _tick_get_32();
+	while (tick == _tick_get_32()) {
+	}
+
+	/* compensate for the extra tick we just waited */
+	++tick;
+
+	k_sleep(SLEEP_TIME);
+
+	tick = _tick_get_32() - tick;
+
+	is_main_thread_ready = 1;
+	k_sem_take(&sem_thread1, K_FOREVER);
+
+	/*
+	 * By design this should be exact, but at least one cycle of
+	 * slop is required experimentally on Qemu.
+	 */
+	zassert_true(tick < SLEEP_TIME || tick > SLEEP_TIME + 1,
+		     "k_sleep() slept for %d ticks, not %d\n",
+		     tick, SLEEP_TIME);
+
+	/*
+	 * Similarly check that thread2 ran for approximately
+	 * SLEEP_TIME. On QEMU, when the host CPU is overloaded, it
+	 * has been observed that the tick count can be missed by 1 on
+	 * either side. Allow for 2 ticks to be sure.  This check is
+	 * only there to make sure that thread2 did run for
+	 * approximately the whole time the main thread was sleeping.
+	 */
+	const int tick_error_allowed = 2;
```

Comment:
no need of variable definition here, the comment is good enough as it is anyway testing app.